### PR TITLE
Adds functionality to read in ini files

### PR DESCRIPTION
**Pull request by adambanker**

### CODE REVIEW

File: paws.common/R/iniutil.R [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Read in values from an ini file
+read_ini <- function(file_name) {
+  if (!file.exists(file_name)) {
+    stop(sprintf("Unable to find file: %s", file_name))
+  }
+  content <- scan(file_name, what = "", sep = "\n", quiet = T)
+  profiles <- list()
+  current_profile <- ""
+  for (i in 1:length(content)) {
+    if (grepl("\\[.*\\]", content[i])) {
```
Comment (davidkretch) [inline paws.common/R/iniutil.R:24]: I would make this `"^\\[.*\\]"` to make sure it only captures lines beginning with `[`.

File: paws.common/R/iniutil.R [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Get the profile name from an ini file
+extract_ini_profile <- function(item) {
+  profile <- gsub("\\[|\\]", "", item)
```
Comment (davidkretch) [inline paws.common/R/iniutil.R:3]: I would make this `"^\\[.*\\]"` to make sure it only captures lines beginning with `[`.

File: paws.common/R/iniutil.R [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  return(profile)
+}
+
+# Get a parameter and its value
+extract_ini_parameter <- function(item) {
+    item_split <- strsplit(item, "=")[[1]]
```
Comment (davidkretch) [inline paws.common/R/iniutil.R:9]: I think this should split at the first `=`, since this might split into more parts if someone has a `=` in the value.

**Comment by davidkretch** [comment]
We should probably also make keys case insensitive, and ignore lines starting with `;`, going by https://en.wikipedia.org/wiki/INI_file.

**Comment by adambanker** [comment]
> We should probably also make keys case insensitive, and ignore lines starting with `;` or `#`, going by https://en.wikipedia.org/wiki/INI_file.

Can you elaborate on what you were thinking for making it case insensitive? This just returns a list so matching keys would be handled elsewhere

**Comment by davidkretch** [comment]
That's a good point. I was thinking of storing all names as lower case, but that is not actually necessary.

**Comment by davidkretch** [comment]
Looks good to me.

**Pull request merged**